guides/hack/21-XHP/16-extending-examples/attribute-transfer.inc.php

```diff
@@ -1,10 +1,14 @@
 <?hh // partial
 
-class :ui-my-good-box extends :x:element {
-  attribute :div; // inherit from attributes from <div>
+use namespace Facebook\XHP\Core as x;
+use type Facebook\XHP\HTML\{div, XHPAttributeClobbering_DEPRECATED};
+
+final xhp class ui_my_good_box extends x\element {
+  attribute :Facebook:XHP:HTML:div; // inherit attributes from <div>
   // Make sure that attributes are transferred automatically when rendering.
-  use XHPHelpers;
-  protected function render(): \XHPRoot {
+  use XHPAttributeClobbering_DEPRECATED;
```

Comment:
This example should probably be replaced with an example of attribute splat

Reply:
How do you use attribute splat to get similar behavior? `<div {...$this}>`? Does it ignore attributes that exist on $this but don't exist on <div>?

Reply:
The spread operator expands to all attributes which are valid on the tag to which the attributes are being transfered to. If the attribute doesn't have a value `null` or `unset` in the transfered from tag, the attribute is not transfered. The relative order of the attribute locations in the transfered to tag does matter. Later occurrences of an attribute override previous ones. For an example see [3.24.0 blog post](https://hhvm.com/blog/2018/01/16/hhvm-3.24.html).

Reply:
Should I recommend using attribute inheritance (`attribute :some:other:element;`) + the splat operator?

I assumed we'd want to move away from attribute inheritance along with attribute clobbering (I removed the inheritance section and said we recommend explicitly listing attributes in the clobbering section), but maybe that's incorrect...?

Reply:
AIUI attribute inheritance is likely staying for as long as attributes remaining a distinct thing to properties (then interfaces can be used) - for now, inheritance + splat seems like the right mix.

Requiring an explicit list will get very verbose - for example, `class`, and `style` are likely to be supported pretty much everywhere

Reply:
I don't quite understand the comment about how interfaces could be used when attributes are backed by properties instead of an associative container. Would you mind clarifying that bit?

Reply:
... TIL that interfaces can't declare properties. I was thinking of this, but it's invalid:

```Hack
interface Foo {
  public string $bar;
}

class HerpDerp implements Foo {}
```

Reply:
Would traits fit the use case? I know they are sometimes frowned upon, but they seems to be up for the task.

I am still a bit unsure about how this could recreate the splat operator.

Reply:
This wouldn't affect the splat operator, it would remove the need for attribute inheritance as a distinct concept though.